pkg/controller.v1beta1/suggestion/suggestion_controller.go

```diff
@@ -191,8 +192,14 @@ func (r *ReconcileSuggestion) ReconcileSuggestion(instance *suggestionsv1beta1.S
 	suggestionNsName := types.NamespacedName{Name: instance.GetName(), Namespace: instance.GetNamespace()}
 	logger := log.WithValues("Suggestion", suggestionNsName)
 
-	// If ResumePolicy = FromVolume volume is reconciled for suggestion
-	if instance.Spec.ResumePolicy == experimentsv1beta1.FromVolume {
+	// Check if algorithm overrides volume reconcile policy
+	suggestionConfigData, err := katibconfig.GetSuggestionConfigData(instance.Spec.Algorithm.AlgorithmName, r.Client)
+	if err != nil {
+		return err
+	}
+
+	// If ResumePolicy = FromVolume (or overriden), volume is reconciled for suggestion
+	if suggestionConfigData.VolumeForceMount || instance.Spec.ResumePolicy == experimentsv1beta1.FromVolume {
```

Comment:
I think we need to identify more places where we check if Suggestion has volume and we need to do something.
e.g. Controller deletes volume when Experiment is finished: https://github.com/kubeflow/katib/blob/3f1eba01d6f6b1520d81b7c71dbfb57c143b62d9/pkg/controller.v1beta1/experiment/experiment_controller_util.go#L168


Maybe we can update [the Suggestion CR's API with `volumeAttached: bool` in addition to `resumePolicy`](https://github.com/kubeflow/katib/blob/3f1eba01d6f6b1520d81b7c71dbfb57c143b62d9/pkg/apis/controller/suggestions/v1beta1/suggestion_types.go#L41).
So we can do appropriate changes based on this spec setting.
We [are creating Suggestion CR](https://github.com/kubeflow/katib/blob/3f1eba01d6f6b1520d81b7c71dbfb57c143b62d9/pkg/controller.v1beta1/experiment/suggestion/suggestion.go#L73) before reconcile Suggestion, so [you don't need to get Katib Config](https://github.com/kubeflow/katib/blob/3f1eba01d6f6b1520d81b7c71dbfb57c143b62d9/pkg/controller.v1beta1/suggestion/suggestion_controller.go#L196) before reconcile Suggestion volume.

What do you think about this API change @tenzen-y @johnugeorge @a9p ?

Reply:
What are the conditions under which katib-controller sets `volumeAttached` to true? Do you plan to do only when `Spec.ResumePolicy == experimentsv1beta1.FromVolume`? Or also check whether `status.phase == Bound` and so on in PVC?

Reply:
One more case where it would be true is where the service needs a volume independent of ResumePolicy (this is what I think the original issue was pointing at separating)